### PR TITLE
Fix #53: Test server: endpoint for online signature calculation

### LINKED ISSUE #53: Test server: endpoint for online signature calculation
Implement test server endpoint which calculates PowerAuth online signature for an activation. The calculated signature value can be used in tests by calling PowerAuth server or PowerAuth cloud signature verification endpoints.

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

File: powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/ResultStatusService.java [hunk 1]
```diff
@@ -116,4 +133,11 @@ private Long getLongValue(JSONObject resultStatusObject, String key) {
         return (Long) resultStatusObject.get(key);
     }
 
+    private TestStatusEntity fetchTestStatus(String activationId) throws ActivationFailedException {
+        final Optional<TestStatusEntity> statusOptional = appStatusRepository.findById(activationId);
+        if (statusOptional.isEmpty()) {
+            throw new ActivationFailedException("Activation with given ID not found: " + activationId);
+        }
+        return statusOptional.get();
```
Comment (banterCZ) [inline powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/ResultStatusService.java:141]:
```suggestion
        return appStatusRepository.findById(activationId).orElseThrow(() ->
                new ActivationFailedException("Activation with given ID not found: " + activationId));
```

File: powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/SignatureService.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+/**
+ * Service for calculating PowerAuth signatures.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@Service
+public class SignatureService extends BaseService {
+
+    private final static Logger logger = LoggerFactory.getLogger(SignatureService.class);
```
Comment (banterCZ) [inline powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/SignatureService.java:51]:
```suggestion
@Slf4j
public class SignatureService extends BaseService {

```

File: powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/SignatureService.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+     * @param request Request for computing a token digest.
+     * @return Response for computing a token digest.
+     * @throws RemoteExecutionException In case remote communication fails.
+     * @throws ActivationFailedException In case activation is not found.
+     * @throws AppConfigNotFoundException In case application configuration is not found.
+     */
+    @SuppressWarnings("unchecked")
+    public ComputeOnlineSignatureResponse computeOnlineSignature(ComputeOnlineSignatureRequest request) throws RemoteExecutionException, ActivationFailedException, AppConfigNotFoundException {
+
+        final String applicationId = request.getApplicationId();
```
Comment (banterCZ) [inline powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/SignatureService.java:82]: What about to add some INFO or DEBUG log (with important context) that the method was called?

**Review by banterCZ — APPROVED** [review]
Nice! Approving with little suggestions.

File: powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/controller/SignatureController.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    /**
+     * Compute a PowerAuth signature.
+     * @param request Compute a PowerAuth signature request.
+     * @return Compute a PowerAuth signature response.
+     * @throws RemoteExecutionException In case remote communication fails.
+     * @throws ActivationFailedException In case activation is not found.
+     * @throws AppConfigNotFoundException In case application configuration is not found.
+     */
+    @RequestMapping(value = "compute-online", method = RequestMethod.POST)
```
[the remark this answers is not in the document]
Comment (romanstrobl) [reply powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/controller/SignatureController.java:62]: Sure.

File: powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/ResultStatusService.java (hunk 1 above)
Comment (romanstrobl) [reply powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/ResultStatusService.java:141]: Yes, old code.

File: powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/SignatureService.java (hunk 2 above)
Comment (romanstrobl) [reply powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/SignatureService.java:51]: Sure.

File: powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/SignatureService.java (hunk 3 above)
Comment (romanstrobl) [reply powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/service/SignatureService.java:82]: Let's do this as part of another ticket, the log messages are missing everywhere

File: powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/model/request/ComputeOnlineSignatureRequest.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+package com.wultra.security.powerauth.app.testserver.model.request;
+
+import lombok.Data;
+
+/**
+ * Request for computing a PowerAuth signature.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@Data
```
[the remark this answers is not in the document]
Comment (romanstrobl) [reply powerauth-test-server/src/main/java/com/wultra/security/powerauth/app/testserver/model/request/ComputeOnlineSignatureRequest.java]: Good point.

**Pull request merged**